core/src/main/java/fr/sncf/osrd/conflicts/Conflicts.kt

```diff
@@ -447,6 +452,12 @@ fun mergeMap(
             events.add(Event(EventType.END, conflict.endTime))
         }
 
+        // TODO: accumulate in the for loop below somehow?
+        val conflictReqs = mutableListOf<ConflictRequirement>()
+        for (conflict in conflicts) {
+            conflictReqs.addAll(conflict.requirements);
+        }
```

Comment:
Seems fine to me, accumulating in the loop wouldn't change the result and this is more straightforward.

That's assuming that we *do* want to list every requirement that appear in the conflict and group them in the same list. When two trains are fully overlapping we will end up with the full requirement list of both trains. But it sounds correct in this context.

Reply:
Honestly, filtering by `event.begin` and `event.time` at line 471 when creating the new conflict seems fine. So if you wanna diminish the duplicated requirements try that, and test it to check that we still have the full list of conflict requirements over all the created objects.

You should probably move line 458 to line 453, it's the same for loop.

Reply:
Only thing left is a unit test then, thx for that :)